src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -142,4 +145,46 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses a {@code String amount} into an {@code Amount}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code amount} is invalid.
+     */
+    public static Amount parseAmount(String amount) throws ParseException {
+        requireNonNull(amount);
+        String trimmedAmount = amount.trim();
+        double value;
+        try {
+            value = Double.parseDouble(trimmedAmount);
+        } catch (NumberFormatException e) {
+            throw new ParseException(Amount.MESSAGE_CONSTRAINTS);
+        }
+        if (!Amount.isValidAmount(Double.parseDouble(amount))) {
+            throw new ParseException(Amount.MESSAGE_CONSTRAINTS);
+        }
+        return new Amount(value);
+    }
+
+    /**
+     * Parses {@code List<String> amounts} into a {@code List<Amount>}.
+     */
+    public static List<Amount> parseAmounts(List<String> amounts) throws ParseException {
+        requireNonNull(amounts);
+        final ArrayList<Amount> amountList = new ArrayList<>();
```

Comment:
why is this final ah

Reply:
doesn't really matter, it's there only cause i copy pasted from the `Set<Tag>` one